circular-buffer/circular_buffer_test.rb

```diff
@@ -86,6 +86,26 @@ def test_overwriting_oldest_item_in_a_full_buffer
     assert_raises(CircularBuffer::BufferEmptyException) { buffer.read }
   end
 
+  def test_forced_writes_of_nil_should_not_occupy_buffer
+    skip
+    buffer = CircularBuffer.new(2)
+    (1..2).each { |i| buffer.write String(i) }
+    buffer.write! nil
```

Comment:
I had the a similar thought. It's notable that all of the test cases only use strings as values for the buffer. If the intent is a circular buffer for strings only, then probably instead of silently failing on a `nil` write, the buffer should throw an exception on any non-string value.

If instead the intent is a buffer that can accept arbitrary objects, then `nil` should be a valid thing to write. Of course, that would make the exercise a notch more difficult since the simple approach of using a `nil` filled array wouldn't work anymore, but it would also be more general an correct.

Since the buffer is suppose to raise an error if you try to read from it empty, there's no scenario where `nil` is used to represent an empty value or anything like that.